### PR TITLE
Limit the number of compile threads on low memory systems

### PR DESCRIPTION
**Pull request by jonapap**
Fixes: #153

### CODE REVIEW

**Comment by ajayyy** [comment]
reason this is draft? does it still have to be tested on a pi?

**Comment by jonapap** [comment]
> reason this is draft? does it still have to be tested on a pi?

Yeah. Need to fine-tune the number of allowed threads. Ideally, we should make it as high as possible without triggering a crash.

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,12 @@ cmake_minimum_required(VERSION 3.12)
 
 project(RocketCode2020 C CXX)
 
+cmake_host_system_information(RESULT _memfree QUERY AVAILABLE_PHYSICAL_MEMORY)
+set_property(GLOBAL PROPERTY JOB_POOLS four_jobs=4)
+if (_memfree LESS 1000)
```
Comment (ajayyy) [inline CMakeLists.txt:7]: Should this be raised to 2gb?

Comment (jonapap) [reply CMakeLists.txt:7]: The RPi we are currently using has a limit of 1GB. Did we have trouble with the other RPi?

**Review by ajayyy — APPROVED** [review]

**Pull request merged**